src/Tools/dotnet-monitor/EndpointInfo/ServerEndpointInfoSource.cs

```diff
@@ -302,5 +304,47 @@ private async Task<bool> CheckEndpointAsync(EndpointInfo info, CancellationToken
 
             return true;
         }
+
+        private IDisposable SetupDiagnosticPortWatcher()
+        {
+            // If running on Windows, a named pipe is used so there is no need to watch it.
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return null;
+            }
+
+            FileSystemWatcher watcher = null;
+            try
+            {
+                watcher = new(Path.GetDirectoryName(_portOptions.EndpointName));
+                void onDiagnosticPortAltered()
+                {
+                    _logger.DiagnosticPortAlteredWhileInUse(_portOptions.EndpointName);
+                    try
+                    {
+                        watcher.EnableRaisingEvents = false;
+                    }
+                    catch
+                    {
+                    }
+                }
+
+                watcher.Filter = Path.GetFileName(_portOptions.EndpointName);
+                watcher.NotifyFilter = NotifyFilters.FileName;
+                watcher.Deleted += (_, _) => onDiagnosticPortAltered();
+                watcher.Renamed += (_, _) => onDiagnosticPortAltered();
+                watcher.Error += (object _, ErrorEventArgs e) => _logger.DiagnosticPortWatchingFailed(_portOptions.EndpointName, e.GetException());
+                watcher.EnableRaisingEvents = true;
+
+                return watcher;
```

Comment:
This now is returning a disposed object. You'll have to store it in a temporary local first, set `watcher` to `null`, and then return the temporary local.

Reply:
Oh, never mind. I read the catcher below as a finally, so this is good.